Extension/src/LanguageServer/configurations.ts

```diff
@@ -1669,13 +1623,29 @@ export class CppProperties {
         });
     }
 
+    public checkCompileCommands(): void {
+        // Check for changes in case of file watcher failure.
+        const compileCommandsFile: string | undefined = this.CurrentConfiguration?.compileCommands;
+        if (!compileCommandsFile) {
+            return;
+        }
+        fs.stat(compileCommandsFile, (err, stats) => {
+            if (err) {
+                if (err.code === "ENOENT" && this.compileCommandsFile) {
+                    this.onCompileCommandsChanged(compileCommandsFile);
+                    this.compileCommandsFile = null; // File deleted
+                }
+            } else if (stats.mtime > this.compileCommandsFileWatcherFallbackTime) {
+                this.compileCommandsFileWatcherFallbackTime = new Date();
```

Comment:
I think this should be moved into this.onCompileCommandsChanged, which matches what is done with configFileWatcherFallbackTime .